libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -2946,6 +2946,11 @@ def add_apperiph_defaults(f):
 #define HAL_EXTERNAL_AHRS_ENABLED 0
 #endif
 
+// disable RC_Channels library:
+#ifndef AP_RC_CHANNEL_ENABLED
```

Comment:
I'm not a big fan of putting conditional compile options in using the chibios script. The script should be able to tell whether things are necessary or not and put them in or not. I think in general conditional compiles should go in a header someplace where they are much easier to maintain

Reply:
On the second point I strongly disagree.  Some of the monstrosities we had when determining compilation in the headers were astonishingly bad.  They still exist, but are slowly being weeded out.  What you're suggesting here is "do away with the hwdef system"!

On the first point - these are simply defaults which Periph has.  Libraries should not know they're doing a periph build in the same way they don't know they're doing a Helicopter build.  So here we have a case where I want to use the RC Protocols on Periph but do not want the RC_Channels library - there's some coupling because we were a little... pragmatic... when changing the behaviour of the RCProtocol library, so here I'm coping with it with `#ifdef`s.

Reply:
we could have a libraries/AP_HAL_ChibiOS/iomcu_config.h and libraries/AP_HAL_ChibiOS/AP_Periph_config.h ?